spec/routine_spec.rb

```diff
@@ -116,4 +116,41 @@ def exec
     end
   end
 
+  it 'does not mess up results on a transaction retry' do
+    # To get the transaction to retry, we need to raise an exception the first time
+    # through execution, after an output has been set in a nested routine and
+    # translated to the parent routine
+
+    stub_const 'RaiseTransactionIsolationConflict', Class.new
```

Comment:
Can we change the name to `NestedRoutine` or something?  It doesn't raise the transaction isolation conflict error anymore :smile: